docs/shaper.md

```diff
@@ -0,0 +1,20 @@
+## GraphQL::Stitching::Shaper
+
+The `Shaper` takes the raw output generated by the `GraphQL::Stitching::Executor` and does the final shaping and
+cleaning. It removes data that was added while building the result, it also handles cleaning up violations that can
+only occur at the end, such as bubbling up null violoations.
+
+See the [Executor](./docs/executor.md)
+
+```ruby
+raw_result = GraphQL::Stitching::Executor.new(
+  supergraph: supergraph,
+  plan: plan.to_h,
+  variables: variables,
+).perform(document)
+
+final_result = GraphQL::Stitching::Shaper.new(
+  supergraph: supergraph,
+  document: document,
+  raw: raw_result).perform!
```

Comment:
@gmac I expect `.perform!` to evoke a strong reaction. It really can not be re-run. The state it is maintaining is not just the raw data, but errors as well. Unless you see value in caching this MethodObject I would suggest we create, run, then destroy it.